GDAS-validation/setup_workspace.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+# setup_workspace.sh
+# this script:
+# - creates working directory
+# - clones global-workflow
+# - clones GDASApp
+# - clones GSI
+# - builds GDASApp + GSI
+# - creates an experiment for a specified cycle/period
+# - links input background files
+
+usage() {
+  set +x
+  echo
+  echo "Usage: $0 [-c] [-b] [-s]"
+  echo
+  echo "  -c  clone necessary repositories"
+  echo "  -b  build GDASApp and GSI"
+  echo "  -s  setup default experiment"
+  echo "  -h  display this message and quit"
+  echo
+  exit 1
+}
+
+clone=NO
+build=NO
+setup=NO
+
+while getopts "cbsh" opt; do
+  case $opt in
+    c)
+      clone=YES
+      ;;
+    b)
+      build=YES
+      ;;
+    s)
+      setup=YES
+      ;;
+    h|\?|:)
+      usage
+      ;;
+  esac
+done
+
+#--------------- User modified options below -----------------
+
+machine=${machine:-orion}
+
+if [ $machine = orion ]; then
+  workdir=/work2/noaa/da/$LOGNAME/gdas-validation
+  ICSDir=/work2/noaa/da/cmartin/UFO_eval/data/para/output_ufo_eval_aug2021
+elif [ $machine = hera ]; then
+  workdir=/scratch1/NCEPDEV/stmp2/$LOGNAME/gdas-validation
+  ICSDir=/scratch1/NCEPDEV/da/Cory.R.Martin/blah/blah
+else
+   echo "Machine " $machine "not found"
+   exit 1
+fi
+
+EXPNAME="gdas_eval_satwind"
+mydir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+
+#-------------- User should not modify below here ----------
+mkdir -p $workdir
+
+#--- clone repositories ---
+if [ $clone = "YES" ]; then
+  cd $workdir
+  echo "Cloning global-workflow at $workdir/global-workflow"
+  git clone https://github.com/noaa-emc/global-workflow.git
+  cd global-workflow/sorc
+  ./checkout.sh -g
```

Comment:
Since we remove the g-w cloned `gsi_enkf.fd`, we could drop the `-g` option and let `checkout.sh` only clone non-GSI and GDASApp repos.

Reply:
I thought about this, but there are the separate gsi_utils and gsi_monitor repositories that the -g flag checks out. I don't think we need the latter but the former may be needed.

Reply:
I missed this.  Good catch.  We'll keep `-g`